packages/react-instantsearch/src/core/createInstantSearchServer.js

```diff
@@ -0,0 +1,136 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { version } from '../../package.json';
+import algoliasearchHelper, {
+  SearchResults,
+  SearchParameters,
+} from 'algoliasearch-helper';
+import ReactDom from 'react-dom/server';
+import { getIndex, hasMultipleIndex } from './indexUtils';
+import { isEmpty } from 'lodash';
+import cis from './createInstantSearch';
+
+const searchParameters = [];
```

Comment:
As I'm aware of, everything is sequential so shouldn't. I tried it with several use of <InstantSearch/> (use case with several instance in the same page).